corehq/motech/fhir/models.py

```diff
@@ -115,6 +115,12 @@ class FHIRResourceProperty(models.Model):
     # do what you need.
     value_source_config = JSONField(null=True, blank=True, default=None)
 
+    def __str__(self):
+        jsonpath = self.value_source_jsonpath
+        if jsonpath.startswith('$.'):
+            jsonpath = jsonpath[2:]
+        return f'{self.resource_type.name}.{jsonpath}'
```

Comment:
I remember once investigating an issue where someone had changed the `__repr__` (or possibly `__str__`) of an object for easier debugging, but that ended up changing user-facing behavior.  I think it was something like:
```python
class Case:
    def __repr__(self):
        return self.name
```
Then in the terminal all you see is "ethan", with no indication of the object type.  The tricky part was that any code that uses this repr probably looks like:
```python
print("hello, {}!".format(case))
```
It'd be very hard to trace down places in the code that depend on that `__repr__` definition.  The alternative would be to make a property like `display_name` and reference that rather than depending on string formatting to call `__str__` or `__repr__` for you.

Anyways, not sure that's quite what's going on here, but I figured I'd flag in case that resonates.  As this is an admin view, the rules are a little different.

Reply:
Thanks. Yeah, that is an important warning!

The Django admin site uses `str()` for headings or titles. The default value for a model is "\<ModelClass\> object (\<id\>)" (e.g. "CaseProperty object (19)") which is unhelpful when you are trying to see what CaseProperty instance you want to edit or select.